stemcell_builder/stages/system_grub/apply.sh

```diff
@@ -9,28 +9,32 @@ source $base_dir/lib/prelude_apply.bash
 
 # Install grub or grub2 (check existence of classic grub package first because Ubuntu Trusty has a transitional grub2 dummy package)
 
-if pkg_exists grub; then
-  pkg_mgr install grub
-elif pkg_exists grub2; then
+# Install grub
+if [ `uname -m` == "ppc64le" ]; then
+  # ppc64le uses grub2
   pkg_mgr install grub2
 else
-  echo "Can't find grub or grub2 package to install"
-  exit 2
+  pkg_mgr install grub
 fi
 
-if [ -d $chroot/usr/lib/grub/x86* ] # classic GRUB on Ubuntu
+
+if [ -f $chroot/etc/debian_version ] # Ubuntu
```

Comment:
shouldnt be checking for ubuntu vs centos. we support other os-es like photon and rhel. what can we do to keep this file as generic as possible?